### PR TITLE
doc: rewrite concepts section

### PR DESCRIPTION
**Pull request by coltmcnealy-lh**
- Enforce that every page in the `/docs/concepts/` on our site corresponds to a `MetadataGetable` or a `GlobalGetable`.
- Makes it easier to find info about how to build workflows in the `/docs/developer-guide/` rather than having to look both in developer guide and `/docs/concepts`
- Makes pipeline fail if broken achor or markdown links are found
- Removes LH for Kubernetes (Formerly LH Platform) docs as they are not for our open-source library.

### CODE REVIEW

File: docs/docs/01-overview.md [hunk 1]
```diff
@@ -118,9 +125,27 @@ In the past, we described LittleHorse as a "microservice orchestration engine" b
...
+The way to integrate LittleHorse with a monolithic app server (for example, Spring Boot) is:
+
+1. The app server upserts the `WfSpec` upon startup via `rpc PutWfSpec`.
+2. Web endpoints (eg. REST, GRPC) on the app server result in calls to LittleHorse to run a `WfRun`.
+3. The app server process also runs Task Workers to implement the tasks in the workflow.
+
 ### API Management System
 
-LittleHorse is not an API Management System like Apigee or Kong.
+LittleHorse is not an API Management System like Apigee or Kong. Those systems define
```
Comment (Snarr) [inline docs/docs/01-overview.md:148]: Define what 🤔

File: docs/docs/04-concepts/30-advanced/10-child-workflows.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Child Workflows
+
+A Child Workflow allows you to
```
Comment (Snarr) [inline docs/docs/04-concepts/30-advanced/10-child-workflows.md:3]: A Child Workflow allows you to launch a new WfRun that is tied to a Parent WfRun. This is similar to the concept of Child Process in operating systems, where a Parent Process spawns a Child Process and can pass down information to it at runtime.

Is this a good start to this page? If so I can continue, and if not, let's discuss this concept more.

File: docs/docs/04-concepts/30-advanced/05-workflow-lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Workflow Lifecycle
```
Comment (Snarr) [inline docs/docs/04-concepts/30-advanced/05-workflow-lifecycle.md:1]: Don't forget about these pages. Maybe we could add an 'UNDER CONSTRUCTION' flag to any page that is in progress.

**Review by Snarr — CHANGES_REQUESTED** [review]
The docs overhaul looks amazing. I merged some of my own changes and left a few comments about unfinished sections.

**Comment by coltmcnealy-lh** [comment]
This is ready now

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -23,7 +23,10 @@ module.exports = {
   // projectName: 'docusaurus', // Usually your repo name.
 
   onBrokenLinks: "throw",
-  onBrokenMarkdownLinks: "warn",
+
+  // Colt gets very grumpy when links don't work.
```
Comment (Snarr) [inline docs/docusaurus.config.js:27]: Nice

**Review by Snarr — APPROVED** [review]
Looks great 👍 This is a great kickoff to our new docs improvements.

**Pull request merged**